shared_model/interfaces/iroha_internal/transaction_sequence.cpp

```diff
@@ -19,11 +20,49 @@ namespace shared_model {
         const validation::TransactionsCollectionValidator<TransactionValidator,
                                                           OrderValidator>
             &validator) {
-      auto answer = validator.validatePointers(transactions);
-      if (answer.hasErrors()) {
-        return iroha::expected::makeError(answer.reason());
+      std::unordered_map<interface::types::HashType,
+                         std::vector<std::shared_ptr<Transaction>>,
+                         interface::types::HashType::Hasher>
+          extracted_batches;
+      std::vector<TransactionBatch> batches;
+
+      auto transaction_validator = validator.getTransactionValidator();
+
+      auto insert_batch =
+          [&batches](const iroha::expected::Value<TransactionBatch> &value) {
+            batches.push_back(value.value);
+          };
+
+      validation::Answer result;
+      for (const auto &tx : transactions) {
+        if (auto meta = tx->batchMeta()) {
+          auto hashes = meta.get()->transactionHashes();
+          auto batch_hash = TransactionBatch::calculateReducedBatchHash(hashes);
+          extracted_batches[batch_hash].push_back(tx);
+        } else {
+          TransactionBatch::createTransactionBatch(tx, transaction_validator)
+              .match(insert_batch, [&tx, &result](const auto &err) {
+                result.addReason(
+                    std::make_pair(std::string("Transaction reduced hash: ")
```

Comment:
The error doesn't seem meaningful maybe explain it more?

Reply:
Error is kept in the err variable